paddle/fluid/operators/elementwise/elementwise_op.h

```diff
@@ -319,16 +319,7 @@ class ElementwiseOpGrad : public framework::OperatorWithKernel {
         ctx, framework::GradVarName("Out"));
 
 #ifdef PADDLE_WITH_MKLDNN
-    // If broadcasting is needed, use native implementation
```

Comment:
So checking these conditions is no longer needed?

Reply:
Apparently it should work.  FWD kernel is more limited by how broadcasting can be done, but here in BWD it is just reduction so according to oneDNN manual it should work  with reduction on any axes. So GetExpectedKernelType of BWD should match FWD. And this is what this PR is doing.